### PR TITLE
construct Transport explicitly rather than copying to make go vet happy

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: go/libkb/client.go [hunk 1]
```diff
@@ -138,7 +139,15 @@ func NewClient(e *Env, config *ClientConfig, needCookie bool) *Client {
...
-	xprt := *(http.DefaultTransport.(*http.Transport)) // shallow copying
+	xprt := http.Transport{ // from http.DefaultTransport
+		Proxy: http.ProxyFromEnvironment,
+		Dial: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}).Dial,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
```
Comment (patrickxb) [inline go/libkb/client.go]: Do we lose anything by not using a shared transport between client connections?

Comment (songgao) [inline go/libkb/client.go]: It looks like by not sharing a transport, we lose cached connections between different `Client` instances. Although this is still a temporary solution. When go1.7 is released (and after we switching to it),  we can revert to the old logic where `Transport` is only [created under certain conditions](https://github.com/keybase/client/commit/2f5ea3745eed0095f5c844482bda79c5c3fc101d#diff-d58cb89c13006e531a58d82658855f64L141).

**Comment by patrickxb** [comment]
LGTM.  Can you add a comment that we should switch back after 1.7?

**Comment by songgao** [comment]
Cool; Thanks Patrick!

**Pull request merged**